Sources/OktaIdxAuth/Internal/Requests/SocialAuth.swift

```diff
@@ -0,0 +1,136 @@
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import AuthenticationServices
+import Foundation
+import OktaIdx
+
+extension OktaIdxAuth.Implementation.Request {
+    @available(iOSApplicationExtension 13.0, *)
+    class SocialAuthenticate: Request<Response>, OktaIdxAuthRemediationRequest {
+        private var webAuthSession: ASWebAuthenticationSession?
+        private let options: OktaIdxAuth.SocialAuth.Options
+        
+        init(options: OktaIdxAuth.SocialAuth.Options, completion: OktaIdxAuth.ResponseResult<Response>?) {
+            self.options = options
+            
+            super.init(completion: completion)
+        }
+        
+        func send(to implementation: OktaIdxAuth.Implementation, from response: IDXClient.Response) {
+            guard let authURL = response.remediation?[.redirectIdp]?.href,
+                  let scheme = URL(string: implementation.client.configuration.redirectUri)?.scheme else
+            {
+                return
+            }
+            
+            self.webAuthSession = ASWebAuthenticationSession(url: authURL, callbackURLScheme: scheme) { (callbackURL, error) in
+                guard error == nil, let callbackURL = callbackURL else {
+                    return
+                }
+                
+                let result = implementation.client.redirectResult(redirect: callbackURL)
+                
+                switch result {
+                case .authenticated:
+                    implementation.client.exchangeCode(redirect: callbackURL) { (token, error) in
+                        if let error = error {
+                            implementation.fail(with: error)
+                            
+                            self.completion?(nil, error)
+                        } else if let token = token {
+                            implementation.delegate?.didSucceed(with: token)
+                            
+                            self.completion?(T(status: .success,
+                                               token: token,
+                                               context: nil,
+                                               additionalInfo: nil),
+                                             nil)
+                        }
+                    }
+                    
+                case .remediationRequired:
+                    implementation.client.introspect { (response, error) in
+                        if let error = error {
+                            implementation.fail(with: error)
+                            
+                            self.completion?(nil, error)
+                        } else if let _ = response {
+                            // Not sure what should be done here
+                        }
+                    }
+                case .invalidContext, .invalidRedirectUrl:
+                    return
```

Comment:
Lets call fatalError in this case, to ensure the developer's completion blocks get called.

Reply:
Done